java/ql/src/semmle/code/java/frameworks/spring/SpringWebUtil.qll

```diff
@@ -0,0 +1,241 @@
+        "org.springframework.web.util;UriBuilder;true;queryParam;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;queryParam;;;Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;queryParam;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;queryParam;(String,Collection);;Element of Argument[1];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;queryParam;(String,Collection);;Element of Argument[1];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;queryParam;(String,Object[]);;ArrayElement of Argument[1];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;queryParam;(String,Object[]);;ArrayElement of Argument[1];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;queryParamIfPresent;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;queryParamIfPresent;;;Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;queryParamIfPresent;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;queryParamIfPresent;;;Element of Argument[1];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;queryParamIfPresent;;;Element of Argument[1];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;queryParams;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;queryParams;;;MapKey of Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;queryParams;;;MapKey of Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;queryParams;;;Element of MapValue of Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;queryParams;;;Element of MapValue of Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;replacePath;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;replacePath;;;Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;replacePath;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;replaceQuery;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;replaceQuery;;;Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;replaceQuery;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParam;;;Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParam;;;Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParam;;;Argument[0];Argument[-1];taint",
```

Comment:
Fluent methods can be represented simpler. If the library sees a value-step from arg to return (signifying a fluent method if the arg is `Argument[-1]`) and a taint/value-step ending in that same argument, then the library will automatically add the step to the return value.
```suggestion
        "org.springframework.web.util;UriBuilder;true;replaceQueryParam;;;Argument[-1];ReturnValue;value",
        "org.springframework.web.util;UriBuilder;true;replaceQueryParam;;;Argument[0];Argument[-1];taint",
```

Reply:
Ah, yeah, I forgot to update this PR.